pkg/trace/otel.go

```diff
@@ -224,6 +225,17 @@ func (t *otelTracer) addInfo(ctx context.Context, args ...log.Marshaler) {
 		for _, arg := range args {
 			kvs := marshalToKeyValue(arg)
 			span.SetAttributes(kvs...)
+
+			// setting the code will cause error spans to be called out specifically
```

Comment:
I was digging into the docs on this `SetStatus` thingy and I found this: https://opentelemetry.io/docs/instrumentation/go/manual/#record-errors.

Should we be doing `span.SetError(v.Error)` here?

Reply:
I think you mean both

```
	span.SetStatus(codes.Error, v.Error)
	span.RecordError(v)
```

I think yes? that does seem better. IIRC, and it's been a while since i experimented with this, RecordError will just set the error attribute in the span, which might clober what's done above, which is probably fine?

Reply:
hokay, i looked into this-- it adds an event, seems reasonable to me https://github.com/open-telemetry/opentelemetry-go/blob/9f42a81a5f8b4faf5689b72f9e4093361d8e5885/sdk/trace/span.go#L369